### PR TITLE
Better keyvault set secret - handle soft-deleted secrets

### PR DESCRIPTION
**Pull request by egil**
This addition adds extra logic to set_KeyVaultSecret which enables it to handle soft deleted secrets automatically by first recovering the secret, and then setting the secret again, if it is different from the recovered secret.

This PR includes a few extra key vault helper functions.

### CODE REVIEW

**Review by lupusbytes — DISMISSED** [review]

File: azure-cli/keyvault/recover_KeyVaultSecret.ps1 [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    --name $secretName
+
+  Throw-WhenError -output $output
+
+  while ($true) {
+    $err = $( $output = az keyvault secret show `
+        --name $secretName `
+        --vault-name $keyVaultName `
+        --query value `
+        --output tsv ) 2>&1
```
Comment (perkops) [inline azure-cli/keyvault/recover_KeyVaultSecret.ps1:25]: What does this do?

Comment (egil) [reply azure-cli/keyvault/recover_KeyVaultSecret.ps1:25]: This is a trick to capture both normal and error output in separate variables such that we can handle the different error/success cases.

File: azure-cli/keyvault/set_KeyVaultSecret.ps1 [hunk 2]
```diff
@@ -29,37 +29,71 @@ function Set-KeyVaultSecret {
...
-    [Parameter(Mandatory=$true)]
+    [Parameter(Mandatory = $true)]
     [ValidateNotNullOrEmpty()]
     [string]
     $secretPlain
   )
 
-  $output = Get-KeyVaultSecret -keyVaultName $keyVaultName -secretName $secretName
+  . "$PSScriptRoot\get_KeyVaultSecret.ps1"
+  . "$PSScriptRoot\recoverKeyVaultSecret.ps1"
```
Comment (lupusbytes) [inline azure-cli/keyvault/set_KeyVaultSecret.ps1]: invalid filename. should be recover-KeyVaultSecret.ps1

File: azure-cli/keyvault/set_KeyVaultSecret.ps1 [hunk 3]
```diff
@@ -29,37 +29,71 @@ function Set-KeyVaultSecret {
...
+  . "$PSScriptRoot\verifyKeyVaultSecret.ps1"
```
Comment (lupusbytes) [inline azure-cli/keyvault/set_KeyVaultSecret.ps1]: invalid filename. should be verify-KeyVaultSecret.ps1

**Review by lupusbytes — APPROVED** [review]

**Pull request merged**